packages/shopify-api/lib/utils/shop-validator.ts

```diff
@@ -17,7 +20,17 @@ export function sanitizeShop(config: ConfigInterface) {
       `^[a-zA-Z0-9][a-zA-Z0-9-_]*\\.(${domainsRegex.join('|')})[/]*$`,
     );
 
-    const sanitizedShop = shopUrlRegex.test(shop) ? shop : null;
+    const unifiedAdminRegex = new RegExp(
+      `^admin.shopify.com/store/([a-zA-Z0-9][a-zA-Z0-9-_]*)$`,
```

Comment:
Should this take into account `admin.${domains}`? I imagined it would need to consider at least `config.customShopDomains` to work with spin URLs?